ingestion-edge/pubsub_emulator.py

```diff
@@ -59,60 +59,55 @@ def create_server(self):
+        self.topics[request.name] = set()
         return request
 
     def DeleteTopic(self, request, context):  # noqa: D403
         """DeleteTopic implementation."""
         try:
             self.topics.pop(request.topic)
         except KeyError:
-            context.set_code(grpc.StatusCode.NOT_FOUND)
+            context.abort(grpc.StatusCode.NOT_FOUND, "Topic not found")
         return empty_pb2.Empty()
 
     def CreateSubscription(self, request, context):  # noqa: D403
         """CreateSubscription implementation."""
         if request.name in self.subscriptions:
-            context.set_code(grpc.StatusCode.ALREADY_EXISTS)
+            context.abort(grpc.StatusCode.ALREADY_EXISTS, "Subscription already exists")
         elif request.topic not in self.topics:
-            context.set_code(grpc.StatusCode.NOT_FOUND)
-        else:
-            subscription = Subscription()
-            self.subscriptions[request.name] = subscription
-            self.topics[request.topic].add(subscription)
+            context.abort(grpc.StatusCode.NOT_FOUND, "Topic not found")
+        subscription = Subscription()
+        self.subscriptions[request.name] = subscription
+        self.topics[request.topic].add(subscription)
         return request
 
     def DeleteSubscription(self, request, context):  # noqa: D403
         """DeleteSubscription implementation."""
         try:
             subscription = self.subscriptions.pop(request.subscription)
         except KeyError:
-            context.set_code(grpc.StatusCode.NOT_FOUND)
-        else:
-            for subscriptions in self.topics.values():
-                subscriptions.discard(subscription)
+            context.abort(grpc.StatusCode.NOT_FOUND, "Subscription not found")
+        for subscriptions in self.topics.values():
+            subscriptions.discard(subscription)
         return empty_pb2.Empty()
 
     def Publish(self, request, context):
         """Publish implementation."""
         if request.topic in self.status_codes:
-            context.set_code(self.status_codes[request.topic])
-            return pubsub_pb2.PublishResponse()
+            context.abort(self.status_codes[request.topic], "Override")
         if self.sleep is not None:
             time.sleep(self.sleep)
         message_ids = []
         try:
             subscriptions = self.topics[request.topic]
         except KeyError:
-            context.set_code(grpc.StatusCode.NOT_FOUND)
-        else:
-            message_ids = [uuid.uuid4().hex for _ in range(len(request.messages))]
-            for _id, message in zip(message_ids, request.messages):
-                message.message_id = _id
-            for subscription in subscriptions:
-                subscription.published.extend(request.messages)
+            context.abort(grpc.StatusCode.NOT_FOUND, "Topic not found")
+        message_ids = [uuid.uuid4().hex for _ in range(len(request.messages))]
```

Comment:
Style suggestion unrelated to the content of your changes:

```suggestion
        message_ids = [uuid.uuid4().hex for _ in request.messages]
```